sshd/src/main/scala/ammonite/repl/sshd/util/Environment.scala

```diff
@@ -0,0 +1,63 @@
+package ammonite.repl.sshd.util
+
+import java.io.{InputStream, OutputStream, PrintStream}
+
+/**
+ * Container for staging environment important for Ammonite repl to run correctly.
+ * @param thread a thread where execution takes place. Important for restoring contextClassLoader
+ * @param contextClassLoader thread's context class loader. Ammonite repl uses that to load it's classes
+ * @param systemIn System.in value
+ * @param systemOut System.out value
+ * @param systemErr System.err value
+ */
+case class Environment(
+  thread:Thread,
+  contextClassLoader: ClassLoader,
+  systemIn: InputStream,
+  systemOut:PrintStream,
+  systemErr:PrintStream
+) {
+  def this(classLoader: ClassLoader, in: InputStream, out:PrintStream) = this(Thread.currentThread(), classLoader, in, out, out)
```

Comment:
Let's use secondary `apply` methods on the companion object, instead of secondary constructors